Integrate support for the TDS2012B Model

kevink17 opened the pull request:
Options for the devices are stored now in BiDict.
Unavailable options will raise a NotImplementedError.

BMecke committed:
Added additional oscilloscope functions, fixed visa query in x_increment function.

* Added functions to get and set the horizontal scale, record length and presample time.
* Added functions to get and set the sample rate (calculated from the horizontal scale).
* Added functions to get and set the presample ratio (calculated from the presample time).
* Changed visa command in x_increment function.
* _data_stop variable is now set to the record length in the constructor.
* Changed url in setup.py to github project website.

BMecke committed:
Added comments for pre_sample_time and pre_sample_ratio functions.

kevink17 committed:
Integrate support for the TDS2012B Model

Options for the devices are stored now in BiDict.
Unavailable options will raise a NotImplementedError.

BMecke committed:
Fixed bugs in __init__ to work with TBS1072C oscilloscope.
Added connected_devices dict.

BMecke committed:
Merge branch 'kevink17_pr1' into dev

# Conflicts:
#	tektronixosc/oscilloscope.py

BMecke committed:
Fixed several bugs (Most capitalized words are now written in lower case)

BMecke commented:
I have tried to adopt most of the changes (dev branch). Can you test whether your oscilloscope still works?

kevink17 committed:
Reformat code

kevink17 committed:
Adapt the divisions depending on the device

kevink17 committed:
Fix bug
Data source command for the tds200 series only returns a single string.

kevink17 committed:
Add consistency

Replace formats with strings and type casts with the query functions

kevink17 committed:
Fix device specific properties

The trigger level property is added for the tds200 series since it is
not channel specific.

kevink17 committed:
Update README

Add TDS2012 to supported models.

kevink17 commented:
Everything works fine on my end. I have added some tweaks which I have overlooked before.

BMecke commented:
I tested the PR today, it seems like everything is working. Many thanks for your cooperation.
